File: Ex-27/ex_27_java/Triangulo.java
class Triangulo {
    private Ponto vertice1;
    private Ponto vertice2;
    private Ponto vertice3;
    
    // Construtor
    public Triangulo(Ponto v1, Ponto v2, Ponto v3) {
        this.vertice1 = v1;
        this.vertice2 = v2;
        this.vertice3 = v3;
    }

    public void setVertice1(double x, double y){
        vertice1.setPonto(x,y);
    }
    public void setVertice2(double x, double y){
        vertice2.setPonto(x,y);
    }
    public void setVertice3(double x, double y){
        vertice3.setPonto(x,y);
    }

    public Ponto getPVertice1() {
        return vertice1;
    }

    public Ponto getPVertice2() {
        return vertice2;
    }

    public Ponto getPVertice3() {
        return vertice3;
    }

    public boolean testeTriangulo() {
        if (vertice1.getX() == vertice2.getX() && vertice1.getX() == vertice3.getX())
            return false;
        if (vertice1.getY() == vertice2.getY() && vertice1.getY() == vertice3.getY())
            return false;
        return true;
    }

    public double calcDist(Ponto p1, Ponto p2) {
        return Math.sqrt( (p1.getX() - p2.getX()) * (p1.getX() - p2.getX()) + (p1.getY() - p2.getY()) * (p1.getY() - p2.getY()));
    }


    public void imprimirPerimetro() {
        float perimetro = 0;

        if (!testeTriangulo()) {
            System.out.println("Os vértices não formam um triangulo, pois são colineares.\n");
            return;
        }

        perimetro += calcDist(vertice1, vertice2);
        perimetro += calcDist(vertice2, vertice3);
        perimetro += calcDist(vertice1, vertice3);

        System.out.println("O perimetro do triangulo é " + perimetro);
    }
}

File: Ex-27/ex_27_python/Triangulo.py
import math

from Ponto import Ponto


class Triangulo:
    def __init__(self):
        self.vertice1 = Ponto()
        self.vertice2 = Ponto()
        self.vertice3 = Ponto()

    def __del__(self):
        del self.vertice1
        del self.vertice2
        del self.vertice3

    def setVertice1(self, x, y):
        self.vertice1.setPonto(x, y)

    def setVertice2(self, x, y):
        self.vertice2.setPonto(x, y)

    def setVertice3(self, x, y):
        self.vertice3.setPonto(x, y)

    def getPVertice1(self):
        return self.vertice1

    def getPVertice2(self):
        return self.vertice2

    def getPVertice3(self):
        return self.vertice3

    # Calcula e imprime o perimetro do triangulo
    def imprimirPerimetro(self):
        perimetro = 0

        if not self.testeTriangulo():
            print("Os vértices não formam um triangulo, pois são colineares.")
            return

        perimetro += self.calcDist(self.vertice1, self.vertice2)
        perimetro += self.calcDist(self.vertice2, self.vertice3)
        perimetro += self.calcDist(self.vertice1, self.vertice3)

        print("O perimetro do triangulo é " + str(perimetro))

    # Testa se os vertices foram um triangulo
    def testeTriangulo(self):
        if self.vertice1.getX() == self.vertice2.getX() and self.vertice1.getX() == self.vertice3.getX():
            return False
        if self.vertice1.getY() == self.vertice2.getY() and self.vertice1.getY() == self.vertice3.getY():
            return False
        return True

    # Calcula a distancia entre dois pontos
    def calcDist(self, p1, p2):
        return math.sqrt((p1.getX() - p2.getX()) * (p1.getX() - p2.getX()) + (p1.getY() - p2.getY()) * (p1.getY() - p2.getY()))
